worlds/oot_soh/location_access/overworld/thieves_hideout.py

```diff
@@ -3,7 +3,233 @@
 if TYPE_CHECKING:
     from ... import SohWorld
 
+class EventsLocations(str, Enum):
+    TH_1_TORCH_CELL_FREE_1_TORCH_CARPENTER = "TH 1 Torch Cell Free 1 Torch Carpenter"
+    TH_DOUBLE_CELL_FREE_DOUBLE_CELL_CARPENTER = "TH Double Cell Free Double Cell Carpenter"
+    TH_DEAD_END_CELL_FREE_DEAD_END_CARPENTER = "TH Dead End Cell Free Dead End Carpenter"
+    TH_STEEP_SLOPE_CELL_FREE_SLOPE_CARPENTER = "TH Steep Slope Cell Free Slope Carpenter"
+    TH_RESCUED_ALL_CARPENTERS = "TH Rescued All Carpenters"
+
+class LocalEvents(str, Enum):
+    TH_1_TORCH_CELL_COULD_FREE_1_TORCH_CARPENTER = "TH 1 Torch Cell Could Free 1 Torch Carpenter"
+    TH_DOUBLE_CELL_COULD_FREE_DOUBLE_CELL_CARPENTER = "TH Double Cell Could Free Double Cell Carpenter"
+    TH_DEAD_END_CELL_COULD_FREE_DEAD_END_CARPENTER = "TH Dead End Cell Could Free Dead End Carpenter"
+    TH_STEEP_SLOPE_CELL_COULD_FREE_SLOPE_CARPENTER = "TH Steep Slope Cell Could Free Slope Carpenter"
```

Comment:
I'd like to see these renamed to "X_FREED" and be shortened a bit. So `TH_1_TORCH_CELL_COULD_FREE_1_TORCH_CARPENTER` would become `TH_1_TORCH_CELL_CARPENTER_FREED`, for example.

Remember to rename them in the rest of the file afterwards too.